src/main/java/com/matera/cadastrocentral/telephone/TelephoneService.java

```diff
@@ -67,15 +71,29 @@ public void deleteTelephone(UUID telephoneId) {
     public Telephone alterTelephoneByTelephoneId(TelephoneDTO telephone) {
         Optional<Telephone> optionalTelephone = telephoneRepository.findById(telephone.getTelephoneId());
         if (optionalTelephone.isPresent()) {
-            Telephone auxTelephone = optionalTelephone.get();
-            auxTelephone.setClientId(Optional.ofNullable(telephone.getClientId())
-                                    .orElse(optionalTelephone.get().getClientId()));
-            auxTelephone.setTelephoneTypeId(Optional.of(telephone.getTelephoneTypeId())
-                                    .orElse(optionalTelephone.get().getTelephoneTypeId()));
+            return telephoneRepository.save(new Telephone(telephone));
+        } else {
+            throw new TelephoneNotFound();
+        }
+    }
+
+    public Telephone patchTelephoneByTelephoneId(TelephoneDTO telephone) {
+        ClientService clientService = new ClientService(clientRepository, null);
```

Comment:
You can inject the ClientService instead of the ClientRepository in the constructor.

once we have the endpoint path set to "/clients/{clientId}/telephones/{telephoneId}", we will have both clientId and telephoneId, so this will not be needed anymore. You will be able to find the right telephone by both fields

Reply:
Actually I was using clientService to check at the table "client" if that clientId exists before changing it, not to find a telephone. Does it make sense?

Reply:
if there is a telephone a client will always exist :) there are no telephones without client on the database constraint,

but its fine to check the clientId that waym but you can inject the ClientServiceDirectly here since you are using it, like:

```
    @Autowired
    public TelephoneService(TelephoneRepository telephoneRepository) {
    public TelephoneService(TelephoneRepository telephoneRepository, ClientService clientService) {
        this.telephoneRepository = telephoneRepository;
        this.clientService = clientService;
    }

    private final TelephoneRepository telephoneRepository;
    private final ClientService clientService;
```